luminaire/exploration/data_exploration.py

```diff
@@ -1070,7 +1071,18 @@ def stream_profile(self, df, impute_only=False, **kwargs):
                         else self.window_length
                     window_length_list.append(window_length_i)
 
-                window_length = int(np.median(window_length_list))
+                window_length_list = np.array(window_length_list)
+                if np.all(window_length_list == min(window_length_list)):
```

Comment:
Is this like a 2d list? Trying to understand what exactly is the comparison happening here since you are using the same variable.